jobs/resource_job.go

```diff
@@ -321,6 +337,16 @@ func (a JobsAPI) Restart(id string, timeout time.Duration) error {
 func (a JobsAPI) Create(jobSettings JobSettings) (Job, error) {
 	var job Job
 	jobSettings.sortTasksByKey()
+	var gitSource *GitSource = jobSettings.GitSource
+	if gitSource != nil && gitSource.Provider == "" {
+		gitSource.Provider = workspace.GetGitProviderFromUrl(gitSource.Url)
+		if gitSource.Provider == "" {
+			return job, fmt.Errorf("git Source is not empty but Git Provider is not specified and cannot be guessed by url %+v", gitSource)
```

Comment:
%s instead of %+v

Reply:
Would you please explain me why is this preferrable?